### ISSUE TITLE
vid_restart crashes the game

### ISSUE THREAD

**Issue by kharnov** [Unvanquished/Unvanquished#822]
/devmap plat23
/vid_restart

```
>WARNING: CG_ReadNextSnapshot: way out of range, 4109 > 0
]
Program received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fffc0da3700 (LWP 2547)]
0x00007fffb635363f in Rocket_ShowHud (weapon=<optimized out>) at /home/kharnov/Unvanquished/src/cgame/rocket/rocket_hud.cpp:122
122                     for ( RocketHud::iterator it = activeHud->begin(); it != activeHud->end(); ++it )
(gdb) bt
#0  0x00007fffb635363f in Rocket_ShowHud (weapon=<optimized out>) at /home/kharnov/Unvanquished/src/cgame/rocket/rocket_hud.cpp:122
#1  0x00007fffb631ebb8 in CG_SetInitialSnapshot (snap=<optimized out>) at /home/kharnov/Unvanquished/src/cgame/cg_snapshot.cpp:130
#2  0x00007fffb631f08d in CG_ProcessSnapshots () at /home/kharnov/Unvanquished/src/cgame/cg_snapshot.cpp:405
#3  0x00007fffb632573b in CG_DrawActiveFrame (serverTime=serverTime@entry=58575, demoPlayback=demoPlayback@entry=false) at /home/kharnov/Unvanquished/src/cgame/cg_view.cpp:1876
#4  0x00007fffb62c96d7 in VM::<lambda(int, bool)>::operator() (demoPlayback=false, serverTime=58575, __closure=<optimized out>) at /home/kharnov/Unvanquished/src/cgame/cg_api.cpp:90
#5  Util::apply_impl<VM::VMHandleSyscall(uint32_t, Util::Reader)::<lambda(int, bool)>, std::tuple<int&&, bool&&>, 0ul, 1ul> (func=<optimized out>, tuple=<optimized out>)
    at /home/kharnov/Unvanquished/daemon/src/common/Util.h:103
#6  Util::apply<VM::VMHandleSyscall(uint32_t, Util::Reader)::<lambda(int, bool)>, std::tuple<int&&, bool&&> > (func=<optimized out>, tuple=<optimized out>)
    at /home/kharnov/Unvanquished/daemon/src/common/Util.h:108
#7  IPC::detail::HandleMsg<VM::VMHandleSyscall(uint32_t, Util::Reader)::<lambda(int, bool)>, IPC::Message<IPC::Id<0u, 3u>, int, bool>, IPC::Reply<> > (channel=..., func=<optimized out>, reader=...)
    at /home/kharnov/Unvanquished/daemon/src/common/IPC/Channel.h:217
#8  IPC::HandleMsg<IPC::SyncMessage<IPC::Message<IPC::Id<0u, 3u>, int, bool> >, VM::VMHandleSyscall(uint32_t, Util::Reader)::<lambda(int, bool)> > (channel=..., func=<optimized out>,
    reader=<error reading variable: access outside bounds of object referenced via synthetic pointer>) at /home/kharnov/Unvanquished/daemon/src/common/IPC/Channel.h:241
#9  VM::VMHandleSyscall (id=<optimized out>, reader=...) at /home/kharnov/Unvanquished/src/cgame/cg_api.cpp:92
#10 0x00007fffb637f3af in CommonInit (rootSocket=<optimized out>) at /home/kharnov/Unvanquished/daemon/src/shared/VMMain.cpp:66
#11 0x00007fffb637f64b in vmMain (rootSocket=<optimized out>) at /home/kharnov/Unvanquished/daemon/src/shared/VMMain.cpp:104
#12 0x00000000005981af in VM::<lambda()>::operator() (__closure=0x632cd28) at /home/kharnov/Unvanquished/daemon/src/engine/framework/VirtualMachine.cpp:352
#13 std::_Bind_simple<VM::CreateInProcessNativeVM(std::pair<IPC::Socket, IPC::Socket>, Str::StringRef, VM::VMBase::InProcessInfo&)::<lambda()>()>::_M_invoke<> (this=0x632cd28)
    at /usr/include/c++/5/functional:1531
#14 std::_Bind_simple<VM::CreateInProcessNativeVM(std::pair<IPC::Socket, IPC::Socket>, Str::StringRef, VM::VMBase::InProcessInfo&)::<lambda()>()>::operator() (this=0x632cd28)
    at /usr/include/c++/5/functional:1520
#15 std::thread::_Impl<std::_Bind_simple<VM::CreateInProcessNativeVM(std::pair<IPC::Socket, IPC::Socket>, Str::StringRef, VM::VMBase::InProcessInfo&)::<lambda()>()> >::_M_run(void) (this=0x632cd10)
    at /usr/include/c++/5/thread:115
#16 0x00007ffff4656030 in ?? () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#17 0x00007ffff3e676aa in start_thread (arg=0x7fffc0da3700) at pthread_create.c:333
#18 0x00007ffff3b9ceed in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:109
```

**Comment by Viech** [comment]
@DolceTriade, reason for closing?

**Comment by Kangz** [comment]
The issue was fixed in 8012e3e959c3fc34f25ab2dec024ad20760e3366
